### PR TITLE
Mr 26 covariate tabs

### PR DESCRIPTION
**Pull request by BalancedmonkeY**
Addition of the standard Bayesian tabs that contain the treatment comparisons for any pair, and the ranking panels.

Note: there is a known bug in the Radial SUCRA plot where it will sometimes not draw the network (which is the case for the test data sets). This bug has been fixed on the main app, and the fix will be included once the meta-regression feature is released in Beta.

### CODE REVIEW

**Review by tommorris168 — CHANGES_REQUESTED** [review]
No comments on the code, all looks good.

Two issues:

1. In tab 4c-3, the download on the leftmost panel is putting the . in the file name in the wrong place, so the file type is wrong.
2. Fixed effects models for shared or unrelated covariate parameters aren't working. I have checked the previous branch (mr-19) and it was working then, so whatever is wrong has been introduced in this branch.

**Comment by BalancedmonkeY** [comment]
Reminder to myself to also add unit tests for slope/intercept code that I implemented for Janion's regression slope

**Review by tommorris168 — CHANGES_REQUESTED** [review]
Plots download properly now and the fixed effects models work.

File: R/gemtc_analysis.R [hunk 1]
```diff
@@ -116,35 +116,60 @@ RunCovariateModel <- function(data, treatment_ids, outcome_type, outcome, covari
...
   # Create text for random/fixed effect
   model_text <- paste(model$model$linearModel,"effect",sep=" ")
   
-  # Summary of model
-  summary <- summary(model)
+  # Summary of relative effects
+  summary <- summary(rel_eff)
+  
+  # Intercepts (regression)
+  intercepts <- summary$summaries$statistics[1:(nrow(model$model$network$treatments)-1),1]
```
[the remark this answers is not in the document]
Comment (BalancedmonkeY) [reply R/gemtc_analysis.R]: I didn't do it quite the same way as the slops, but I have made it now that the intercepts are grabbed by their name (i.e. presence of 'd.').

File: R/gemtc_analysis.R [hunk 2]
```diff
@@ -116,35 +116,60 @@ RunCovariateModel <- function(data, treatment_ids, outcome_type, outcome, covari
...
   # Table of Model fit stats
   fit_stats <- as.data.frame(summary$DIC)
   
   # Summary sentence of where covariate value has been set for results
   cov_value_sentence <- paste0("Value for covariate ", model$model$regressor$variable, " set at ", cov_value)
   
+  # Obtain slope(s)
+  slope_indices <- grep(ifelse(model$model$regressor$coefficient == "shared", "^B$", "^beta\\[[0-9]+\\]$"), model$model$monitors$enabled)
+  summ <- summary(model)
+  slopes <- summ$summaries$statistics[slope_indices, 1]  / model$model$regressor$scale
```
[the remark this answers is not in the document]
Comment (BalancedmonkeY) [reply R/gemtc_analysis.R]: Added the "Mean" to both slopes and intercepts - thanks.

And yes, the scale calculation was something that Janion picked up.

**Review by tommorris168 — CHANGES_REQUESTED** [review]
Sorry, another change. I've just looked at the forest plots in gemtc and bnma, and they both use the median, not the mean. To be consistent I think we should use the median in the novel plot as well. So we want ...$summaries$quantiles[row_index, "50%"].

**Comment by BalancedmonkeY** [comment]
> Sorry, another change. I've just looked at the forest plots in gemtc and bnma, and they both use the median, not the mean. To be consistent I think we should use the median in the novel plot as well. So we want ...$summaries$quantiles[row_index, "50%"].

Done - and no worries, this does make sense. Also, this may actually help with some of Janion's tests!

**Review by tommorris168 — APPROVED** [review]

**Pull request merged**